### PR TITLE
feat: add supporting for monitoring s3 buckets for events

### PR DESCRIPTION
**Pull request by NuclearRedeye**
An initial stake in the ground more that anything, as there are still some creases that need ironing out. But can monitor a bucket, or buckets, via an SQS queue and internally handle them.

Note that the integration test is disabled for now, as there are configuration concerns that need to be thought through for how they can be automatically run (Jenkins I presume)

### CODE REVIEW

File: .vscode/launch.json [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+      "request": "launch",
+      "name": "Debug Service",
+      "preLaunchTask": "npm: build",
+      "program": "${workspaceFolder}/dist/bin/index.js",
+      "runtimeArgs": ["--experimental-modules"],
+      "args": [],
+      "env": { "AWS_REGION": "us-east-2", "AWS_PROFILE": "test-editor-bucket-monitor" }
+    }
+  ]
+}
```
Comment (ilya-zhuravel) [inline .vscode/launch.json]: shall we add IDE folders to gitignore?

File: src/awsSqsQueueMonitor.ts [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+    super();
+
+    this.queue = queueUrl;
+    this.buckets = buckets;
+    this.monitoring = false;
+
+    this.sqs = new AWS.SQS();
+  }
+
+  async poll() {
```
Comment (ilya-zhuravel) [inline src/awsSqsQueueMonitor.ts:20]: This method seems to be doing quite a few things. Maybe it breaking it down into multiple methods would make it easier to read

File: src/awsSqsQueueMonitor.ts [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+    this.monitoring = true;
+    do {
+      await this.poll();
+    } while (this.monitoring);
+  }
+
+  stop(): void {
+    this.monitoring = true;
+  }
+
```
Comment (ilya-zhuravel) [inline src/awsSqsQueueMonitor.ts:81]: Should the methods below be private?

File: tsconfig.json [hunk 4]
```diff
@@ -8,8 +8,9 @@
     "strict": true,
     "noImplicitAny": true,
     "preserveConstEnums": true,
-    "strictNullChecks": false
+    "strictNullChecks": false,
+    "esModuleInterop": true
   },
   "include": ["src/**/*"],
-  "exclude": ["node_modules", "src/**/*.test.*", "src/**/*.spec.*", "test/**/*"]
+  "exclude": ["node_modules", "src/**/*.test.*", "src/**/*.spec.*", "test_old/**/*"]
```
Comment (ilya-zhuravel) [inline tsconfig.json]: Is that meant to be renamed?

**Review by ilya-zhuravel — CHANGES_REQUESTED** [review]
Some minor changes only. 
Also, what do we do about package-lock.json? It's disabled in texture but enabled here and in editor-demo. I guess we could benefit from having a consistent approach.

File: .vscode/launch.json (hunk 1 above)
Comment (NuclearRedeye) [reply .vscode/launch.json]: Good point, will address...

File: src/awsSqsQueueMonitor.ts (hunk 2 above)
Comment (NuclearRedeye) [reply src/awsSqsQueueMonitor.ts:20]: Agreed. I'm likely to tweak this a bit in the next task, I can address this then.

File: tsconfig.json (hunk 4 above)
Comment (NuclearRedeye) [reply tsconfig.json]: Ah, good spot :-)

**Review by ilya-zhuravel — APPROVED** [review]

**Pull request merged**